src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Report;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy strategy;
+
+    public ShopServiceImpl(OperationStrategy strategy) {
+        this.strategy = strategy;
+    }
+
+    @Override
+    public Report process(List<FruitTransaction> transactions) {
+        Map<String, Integer> fruitQuantityMap = new HashMap<>();
```

Comment:
According to the checklist, use data structures from the Collection framework that are most suitable for your needs. A HashMap is a good choice, but ensure it is the most appropriate for the operations you are performing on it, especially considering the frequency of read/write operations and the need for ordering.